turtle.js

```diff
@@ -378,13 +366,11 @@ function colour(r, g, b, a) { // should this have a `color` alias?
  * @param {number} min
  * @param {number} max
  */
-const random = function(min, max) {
-    return Math.floor(Math.random() * (max - min + 1) + +min);
-};
+const random = (min, max) => Math.floor(Math.random() * (max - min + 1) + +min);
```

Comment:
The extra unary operator `+` for `+min` seems unnecessary.

Reply:
I agree. But that fn can be called by external users, which can pass arbitrary values, so it's a good idea to prevent string-concat by doing an explicit number coercion

Reply:
> , so I thought it was a good idea to prevent string-concat...

I also love to use the unary `+` operator for string-to-number coercions.

However, by safe-guarding the parameter _min_ only and skipping _max_ makes the whole thing incoherent.

You'd also need to apply those same safe-guard string-to-number coercions everywhere within the code.

Reply:
> However, by safe-guarding the parameter _min_ only and skipping _max_ makes the whole thing incoherent.

Yeah, I think we should explicitly coerce both `min` & `max`, to make intent clearer, and to avoid the need for an explanatory comment (this is similar to the self-commenting code philosophy). It could also improve performance by making optimizations easier, because all types would match before any arithmetic is done

> You'd also need to apply those same safe-guard string-to-number coercions everywhere within the code.

Yep, I forgot about other fns lol

> Typescript would catch that much faster than me. :P

lmao, true